pkg/selector/emr.go

```diff
@@ -0,0 +1,342 @@
+package selector
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+// EMR is a Service type for a custom service filter transform
+type EMR struct{}
+
+// Filters implements the Service interface contract for EMR
+func (e EMR) Filters(version string) (Filters, error) {
+	filters := Filters{}
+	instanceTypes, err := e.getEMRInstanceTypes(version)
+	if err != nil {
+		return filters, err
+	}
+	filters.InstanceTypes = &instanceTypes
+	filters.RootDeviceType = aws.String("ebs")
+	filters.VirtualizationType = aws.String("hvm")
+	return filters, nil
+}
+
+// getEMRInstanceTypes returns a list of instance types that emr supports
+func (e EMR) getEMRInstanceTypes(version string) ([]string, error) {
+	instanceTypes := []string{}
+	if version == "" {
+		version = "5.20.0"
+	}
+	versionInt, err := strconv.Atoi(strings.ReplaceAll(version, ".", ""))
+	if err != nil {
+		return instanceTypes, err
+	}
+
+	for _, instanceType := range e.getAllEMRInstanceTypes() {
+		if versionInt >= 5250 {
+			instanceTypes = append(instanceTypes, instanceType)
+		} else if versionInt >= 5200 && versionInt < 5250 {
+			if e.isOnlyEMR_5_25_0_plus(instanceType) {
+				continue
+			}
+			instanceTypes = append(instanceTypes, instanceType)
+		} else if versionInt >= 5150 && versionInt < 5200 {
+			if instanceType == "c1.medium" {
```

Comment:
why call out this type?

Reply:
I just called it that since there was already a string version and this should be the string version w/ no dot as an int for comparisons. Definitely not the cleanest.... maybe a better way is to make a comparison func that accepts the version as a string.

Reply:
used semver lib instead